lib/rucio/daemons/rsedecommissioner/config.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright European Organization for Nuclear Research (CERN) since 2012
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions to manage decommissioning configurations."""
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from rucio.core.rse import add_rse_attribute, get_rse_attribute
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+class DecommissioningStatus(Enum):
+    """Decommissioning status flags."""
+
+    PROCESSING = 'processing'
+    DONE = 'done'
+    SUSPENDED = 'suspended'
+
+
+class InvalidStatusName(Exception):
+    pass
+
+
+def config_to_attr(config: 'dict[str, Any]') -> str:
+    """Form the attribute string from a config dictionary.
+
+    :param config: Decommissioning configuration dictionary.
+    :returns: Comma-separated key-value string encoding the configuration.
+    """
+    attr = f'profile={config["profile"].value}'
+    if config.get('move_dest'):
+        attr += f',move_dest={config["move_dest"]}'
+    attr += f',status={config["status"].value}'
+
+    return attr
+
+
+def attr_to_config(attr: str) -> 'dict[str, Any]':
+    """Form the config dictionary from an attribute string.
+
+    :param attr: Comma-separated key-value string encoding the configuration.
+    :returns: Decommissioning configuration dictionary.
+    """
+    # The decommission attribute is a comma-separated list of key=value settings
+    config = dict(map(lambda s: s.split('='), attr.split(',')))
+    if 'status' not in config:
+        config['status'] = 'processing'
+    try:
+        config['status'] = DecommissioningStatus[config['status'].upper()]
```

Comment:
Something’s amiss here. Maybe you wanted to assign `DecommissioningStatus.PROCESSING` instead of `'processing'`?

Reply:
I was assigning a string to make `config['status']` look like it got a value from the RSE attribute. Then in the next lines the string status would be converted to the Enum. But indeed that is a roundabout way of doing something simple. I rewrote this part to directly assign the Enum value.